bindings/pybind11/tests/test_idyntree_vector_casters.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for idyntree-vector-casters."""
+import unittest
+
+from idyntree_pybind11_test.vector_casters import TestClass
+import numpy as np
+
+
+class TestClassTest(unittest.TestCase):
```

Comment:
[Question] 
I am trying to understand what we are testing with this test case.

What we want is to check that Python List/Numpy objects are converted to the corresponding iDynTree classes and vice versa.
Is this test enough for testing this?

Let's take the first test:
```py
  def test_vector_fix(self):
     obj = TestClass()
     vector = [1., 2., 3.]
     obj.vector_fix = vector
     self.assertEqual(len(obj.vector_fix), 3)
     self.assertEqual(list(obj.vector_fix), list(vector))
```

`obj.vector_fix = vector` tests that the conversion into a fix<3> works.
The other two lines are then testing the conversion back into Python.
But, can we add a test that if we have a C++ method operating on the vector it is happy? e.g. summing 1 on all elements.

Reply:
>  But, can we add a test that if we have a C++ method operating on the vector it is happy? e.g. summing 1 on all elements.

@GiulioRomualdi any toughts? Also for this I think we can merge in this state and eventually propose new PRs.